### ISSUE TITLE
useReadonlyClassProperties doesn't recognise ++var

### ISSUE THREAD

**Issue by hyperknot** [biomejs/biome#6800]
### Environment information

```bash
playground
```

### Rule name

useReadonlyClassProperties

### Playground link

https://biomejs.dev/playground/?lintRules=useReadonlyClassProperties&code=ZQB4AHAAbwByAHQAIABjAGwAYQBzAHMAIABXAGUAYgBEAGIAQgByAGkAZABnAGUAIAB7AAoAIAAgAHAAcgBpAHYAYQB0AGUAIABuAGUAeAB0AEkAZAAgAD0AIAAwAAoACgAgACAAYwBvAG4AcwB0AHIAdQBjAHQAbwByACgAKQAgAHsACgAgACAAIAAgAGMAbwBuAHMAdAAgAGEAIAA9ACAAKwArAHQAaABpAHMALgBuAGUAeAB0AEkAZAAKACAAIAAgACAAYwBvAG4AcwBvAGwAZQAuAGwAbwBnACgAYQApAAoAIAAgAH0ACgB9AAoA

### Expected result

Should not be readonly

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by vladimir-ivanov** [comment]
thanks for reporting the bug.
The biome useReadonlyClassProperties aims to be fully aligned with its eslint equivalent.
As such, constructor is a more special case and behaviour is different compared to other class methods.

Please see:

<img width="527" height="227" alt="Image" src="https://github.com/user-attachments/assets/4e94a808-7c9d-470f-adec-e4bdd8f159db" />

and link to playground here:

However this is the same behaviour as eslint: 
https://typescript-eslint.io/play/#ts=5.8.2&fileType=.ts&code=KYDwDg9gTgLgBAYwDYEMDOa4HVgCMAiuAQlAJYAmA5sHAN4BQccYZAbijDQHagwCS5OAF44ABnqNEELmhhQArghjQAFAEo6kpgmmy4KYXADURmAAtSaAHQ8Q-clqkyISYFaQRKKlGskBfej8gA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0Wls4CGAEwD2TeIXRRe0EdEjgwAXxBKgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

Will close for time being, but if you disagree please reopen and we can continue the discussion?
